services/resetPassword.js

```diff
@@ -0,0 +1,42 @@
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+const userRepository = require('../repositories/user');
+const errors = require('../configuration/errors');
+
+async function showResetPasswordForm(req, res) {
+  const token = req.params.token;
+  try {
+    const decodedToken = jwt.verify(token, process.env.JWT_SECRET);
+    const user = await userRepository.getUserById(decodedToken.userId);
+
+    if (!user) {
+      return res.status(404).send(errors.USER_NOT_FOUND);
+    }
+    res.render('resetPassword', { token });
+  } catch (error) {
+    return res.status(400).send(errors.INVALID_TOKEN);
+  }
+}
+
+async function resetPassword(token, newPassword, confirmPassword) {
+  const decodedToken = jwt.verify(token, process.env.JWT_SECRET);
+  const user = await userRepository.getUserById(decodedToken.userId);
+
+  if (!user) {
+    throw new Error(errors.USER_NOT_FOUND);
+  }
+  if (decodedToken.exp < Date.now() / 1000) {
```

Comment:
I think verify will return false if token is expired. as well as if token is invalide for some other reason.